pyproject.toml

```diff
@@ -72,9 +70,63 @@ push = true
 "example/docs/conf.py" = ['^release = "{pep440_version}"$']
 "src/talondoc/_version.py" = ['^__version__: str = "{pep440_version}"$']
 
-[tool.isort]
-profile = "black"
-line_length = 88
+[tool.ruff]
+line-length = 88
+
+exclude = [".venv", ".tox", "example/community/*"]
+
+[tool.ruff.lint]
+# Add the `line-too-long` rule to the enforced rule set. By default, Ruff omits rules that
+# overlap with the use of a formatter, like Black, but we can override this behavior by
+# explicitly adding the rule.
+# extend-select = ["E501"]
+
+select = [
+  # Recommended Initial
+  "E",   # pycodestyle
+  "F",   # Pyflakes
+  "UP",  # pyupgrade
+  "B",   # flake8-bugbear
+  "SIM", # flake8-simplify
+  "I",   # isort
+
+  # Additional
+  "PTH", # flake8-use-pathlib
+  "C4",  # flake8-comprehensions (C4)
+  "PIE", # flake8-pie (PIE)
+  "PYI", # flake8-pyi (PYI)
+  "RUF", # Ruff Specific Rules
+  "RET", # flake8-return (RET)
+
+  "FLY", # flynt (FLY)
+
+]
+
+ignore = [
```

Comment:
Why did you add this ignore list?

Reply:
I am okay if you want to remove that ignore rule. That rule wants us to use `Path(x).Open(options)` instead of just `open(x, options). I think that for context management with `open(str)` is more concise. There was only a few instances where `with open(x) as y:. That is the symantics that the python docs uses. Most of the instances of calling `open` in the code vase just used a constructed string as the input. I have no issue if you want me to change them to the more verbose option and removing the ignore rule

Reply:
The other parts of the ignore list were added as they conflict with the formatter per https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules